### ISSUE TITLE
An error occurred while running semantic-release: Error: You provided a local path.

### ISSUE THREAD

**Issue by Alorel** [semantic-release/semantic-release#963]
## Current behavior

Running semantic-release on travis-ci.com throws

```
[9:17:13 AM] [semantic-release] › ✖  An error occurred while running semantic-release: Error: You provided a local path.
    at parseUrl (/home/travis/build/Alorel/personal-build-tools/node_modules/parse-url/lib/index.js:48:15)
    at gitUp (/home/travis/build/Alorel/personal-build-tools/node_modules/git-up/lib/index.js:29:18)
    at gitUrlParse (/home/travis/build/Alorel/personal-build-tools/node_modules/git-url-parse/lib/index.js:42:19)
    at module.exports (/home/travis/build/Alorel/personal-build-tools/node_modules/semantic-release/lib/get-git-auth-url.js:49:36)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
Error: You provided a local path.
    at parseUrl (/home/travis/build/Alorel/personal-build-tools/node_modules/parse-url/lib/index.js:48:15)
    at gitUp (/home/travis/build/Alorel/personal-build-tools/node_modules/git-up/lib/index.js:29:18)
    at gitUrlParse (/home/travis/build/Alorel/personal-build-tools/node_modules/git-url-parse/lib/index.js:42:19)
    at module.exports (/home/travis/build/Alorel/personal-build-tools/node_modules/semantic-release/lib/get-git-auth-url.js:49:36)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
```

## Expected behavior

semantic-release works without throwing errors. 

## Environment

- **semantic-release** version: `~15.10.3`
- CI environment: `travis-ci.com`
- Plugins used:
  - `@semantic-release/changelog@~3.0.0`
  - `@semantic-release/exec@~3.3.0`
  - `@semantic-release/git@~7.0.1`
  - `@semantic-release/npm@~5.2.0`
  - [`@alorel-personal/semantic-release`](https://github.com/Alorel/semantic-release-alorel) (modifies travis, coveralls etc badges in the README to reflect the released version).
  - [`@alorel-personal/conventional-changelog-alorel`](https://github.com/Alorel/conventional-changelog-personal) (modified Angular conventional changelog preset)
- **semantic-release** configuration: https://github.com/Alorel/personal-build-tools/blob/09418cdc8c168b2b6fa7dd72fbd78b51614c9bdd/.releaserc.yml
- CI logs: https://travis-ci.com/Alorel/personal-build-tools/jobs/153804317
  - Uses node_js version `stable` (11.0). Reverting to `lts/*` (8.x) has no effect on the issue.
- Diff from previous successful release: https://github.com/Alorel/personal-build-tools/compare/4.1.3...09418cdc8c168b2b6fa7dd72fbd78b51614c9bdd

**Comment by hiradimir** [comment]
I have same issue on circleci


```sh
$ npx semantic-release
npx: installed 780 in 15.47s
[11:34:35 PM] [semantic-release] › ℹ  Running semantic-release version 15.10.5
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[11:34:35 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[11:34:35 PM] [semantic-release] › ✔  Run automated release from branch production
[11:34:35 PM] [semantic-release] › ✖  An error occurred while running semantic-release: Error: You provided a local path.
    at parseUrl (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/parse-url/lib/index.js:48:15)
    at gitUp (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/git-up/lib/index.js:29:18)
    at gitUrlParse (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/git-url-parse/lib/index.js:42:19)
    at module.exports (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/lib/get-git-auth-url.js:49:36)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Error: You provided a local path.
    at parseUrl (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/parse-url/lib/index.js:48:15)
    at gitUp (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/git-up/lib/index.js:29:18)
    at gitUrlParse (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/node_modules/git-url-parse/lib/index.js:42:19)
    at module.exports (/home/circleci/.npm/_npx/53/lib/node_modules/semantic-release/lib/get-git-auth-url.js:49:36)
    at process._tickCallback (internal/process/next_tick.js:68:7)Exited with code 1
```

**Comment by tuckbick** [comment]
I was able to fix this issue by changing the repository url in the package.json from `git` format to `https` format.

**Comment by zanona** [comment]
@tuckbick Yes, however, if you have read/write GitHub deploy keys installed by the CI, it seems that its server is only able to access the repo through `git@github.com:username/repo.git` while `https://github.com/username/repo.git` always asks for a username and password. I personally believe repo-specific deploy keys are much safer than an account personal token.
I have trouble understanding as well why this message is showing up.

**Comment by Alorel** [comment]
@tuckbick An auth error shows up if I <a href="https://github.com/Alorel/personal-build-tools/commit/73bc9582818cd5e6f556e29cee26b287ffd0a37a">change</a> the git repo URL to its http version.

<details>
<summary>Show</summary>
<pre>
[2:03:04 PM] [semantic-release] › ℹ  Running semantic-release version 15.10.5
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@alorel-personal/semantic-release"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/exec"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/exec"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/exec"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[2:03:04 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[2:03:04 PM] [semantic-release] › ✔  Run automated release from branch master
[2:03:07 PM] [semantic-release] › ✖  The command "git push --dry-run http://[secure]@github.com/Alorel/personal-build-tools.git HEAD:master" failed with the error message remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/Alorel/personal-build-tools.git/'
.
[2:03:07 PM] [semantic-release] › ℹ  Start step "fail" of plugin "@semantic-release/github"
[2:03:07 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[2:03:11 PM] [semantic-release] [@semantic-release/github] › ℹ  Created issue #34: https://github.com/Alorel/personal-build-tools/issues/34.
[2:03:11 PM] [semantic-release] › ✔  Completed step "fail" of plugin "@semantic-release/github"
[2:03:11 PM] [semantic-release] › ✖  EGITNOPERMISSION The push permission to the Git repository is required.
semantic-release cannot push the version tag to the branch master on remote Git repository with URL http://[secure]@github.com/Alorel/personal-build-tools.git.
Please refer to the authentication configuration documentation (https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/ci-configuration.md#authentication) to configure the Git credentials on your CI environment and make sure the repositoryUrl (https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/configuration.md#repositoryurl) is configured with a valid Git URL (https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols).
{ SemanticReleaseError: The push permission to the Git repository is required.
    at module.exports (/home/travis/build/Alorel/personal-build-tools/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (/home/travis/build/Alorel/personal-build-tools/node_modules/semantic-release/index.js:75:11)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
  name: 'SemanticReleaseError',
  code: 'EGITNOPERMISSION',
  details:
   '**semantic-release** cannot push the version tag to the branch `master` on remote Git repository with URL `http://[secure]@github.com/Alorel/personal-build-tools.git`.\n\nPlease refer to the [authentication configuration documentation](https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/ci-configuration.md#authentication) to configure the Git credentials on your CI environment and make sure the [repositoryUrl](https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/configuration.md#repositoryurl) is configured with a [valid Git URL](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols).',
  semanticRelease: true }
The command "semantic-release" exited with 1.
</pre>
</details>

<br/>
Either way this error shows there's <i>some</i> sort of regression going on. The auth tokens/environment variables haven't changed

**Comment by pvdlg** [comment]
This is caused by https://github.com/IonicaBizau/git-up/issues/17.

**Comment by ivandov** [comment]
We're hitting this now too in our production build cycle, with no viable workaround found. 

Shouldn't `semantic-release` specify a specific version of `git-up` in its `package.json`?

**Comment by martinmosegaard** [comment]
Hi @pvdlg. May I ask if this could be an opportunity to revisit the caret range dependency on git-url-parse? I am not an expert on npm dependencies and it seems there are different best practices. But using a tilde or pinned dependency might help the stability of semantic-release? Seems this mishap had a real impact on CI/CD environments. Thanks.

**Comment by Alorel** [comment]
This is a temporary workaround idea **that will only be suitable for a small portion of us**, but you can try to "force-lock" git-url-parse to a specific version by:

1. using yarn as the package manager
1. adding a specific version of git-url-parse to your project dependencies
1. using flat mode - this should prompt you to select the version you want to end up with, similar to how bower works.

Just something off the top of my head until a proper fix is pushed.

**Comment by ersel** [comment]
We pinned git-up v2.0.10 as a devDependency in `package.json` for now.

`"git-up": "2.0.10",`

**Comment by pvdlg** [comment]
@martinmosegaard, @Alorel the problem is in `git-up` which is a dependency `git-url-parse`. The specific version of `git-up` is in range of the definition in `git-url-parse` so there is nothing that can be done in semantic-release. Change our dependency definition of `git-url-parse` would not change anything.

**Comment by Alorel** [comment]
@pvdlg I'm aware, which is why I didn't suggest you make a change, this is more for projects that depend on `semantic-release` to lock down on a `git-up` version until a fix is pushed. And I only just noticed that I posted the wrong dependency name in the comment, I'll update it.

Alternatively, people can manually modify their lockfiles as a temp fix. I successfully published a release from [this](https://github.com/Alorel/tslint-rules) repo just a few hours ago; copy over the `semantic-release` version range from package.json and the lockfile data for the relevant dependencies from yarn.lock. Not exactly ideal, but it's something :slightly_smiling_face:
